### PR TITLE
Wallet: Delegation and Staking flows 

### PR DESCRIPTION
**Pull request by michellewong793**

### CODE REVIEW

File: frontend/wallet/src/common/Tc.re [hunk 1]
```diff
@@ -106,6 +106,8 @@ module Option = {
 
   let getExn = x => Belt.Option.getExn(x);
 
+  let flatMap = Belt.Option.flatMap;
```
Comment (Schmavery) [inline frontend/wallet/src/common/Tc.re]: Option.andThen is probably what you want?

File: frontend/wallet/src/render/views/settings/AccountSettings.re [hunk 2]
```diff
@@ -181,6 +200,156 @@ module DeleteButton = {
...
+                       onClick={_ =>
+                         ReasonReact.Router.push(
+                           "/settings/"
+                           ++ PublicKey.uriEncode(publicKey)
+                           ++ "/stake",
+                         )
+                       }
+                     />
+                     <img
+                       src="https://cdn.discordapp.com/attachments/638495089232183306/641796805314478092/OR.png"
```
Comment (Schmavery) [inline frontend/wallet/src/render/views/settings/AccountSettings.re]: oops are we unable to load assets

File: frontend/wallet/src/render/views/settings/DelegationSettings.re [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+                  label="Fee"
+                  value={
+                    feeAmount == Int64.zero ? "" : Int64.to_string(feeAmount)
+                  }
+                  placeholder="0"
+                  onChange={value => {
+                    let serializedValue =
+                      switch (value) {
+                      | "" => Int64.zero
+                      | nonEmpty => Int64.of_string(nonEmpty)
```
Comment (Schmavery) [inline frontend/wallet/src/render/views/settings/DelegationSettings.re]: It looks like non-int fees might crash this?

File: frontend/wallet/src/render/views/settings/DelegationSettings.re [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+         </span>
+         <Spacer height=1. />
+         <div className=Css.(style([display(`flex)]))>
+           <Button label="Cancel" style=Button.Gray onClick={_ => goBack()} />
+           <Spacer width=1. />
+           <Button
+             label="Delegate"
+             style=Button.Green
+             disabled=loading
+             onClick={_ => mutate(~variables, ~refetchQueries=[|"getAccountInfo"|], ()) |> ignore}
```
Comment (Schmavery) [inline frontend/wallet/src/render/views/settings/DelegationSettings.re]: Might want to also refetch queryDelegation to change the header text

File: frontend/wallet/src/render/views/settings/SettingsPage.re [hunk 5]
```diff
@@ -122,14 +129,25 @@ module Version = {
...
+  Account.t = {
+    locked: option(bool),
+    publicKey: PublicKey.t,
+    balance: {. "total": int64},
+  };
+  
 module AccountSettingsItem = {
   [@react.component]
-  let make = (~publicKey) => {
+  let make = (~publicKey, ~account) => {
```
Comment (Schmavery) [inline frontend/wallet/src/render/views/settings/SettingsPage.re]: Does account contain the pubkey?
It also looks pretty similar to the ownedAccounts type, any reason we're not using that?

File: frontend/wallet/src/render/views/settings/DelegationSettings.re (hunk 4 above)
Comment (michellewong793) [reply frontend/wallet/src/render/views/settings/DelegationSettings.re]: 👍 got it!

File: frontend/wallet/src/render/views/settings/SettingsPage.re (hunk 5 above)
Comment (michellewong793) [reply frontend/wallet/src/render/views/settings/SettingsPage.re]: it does- should we just use account instead of passing in both publicKey and account?

Comment (michellewong793) [reply frontend/wallet/src/render/views/settings/SettingsPage.re]: changed it!

**Comment by bkase** [comment]
That's a flake on bootstrap; I'm landing this as it doesn't touch ocaml and this one passed last time

**Pull request merged**